### PR TITLE
Improve readability of job controller unit tests

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind cleanup

#### Does this PR introduce a user-facing change?

```release-note
NONE

```

### CODE REVIEW

File: pkg/controller/jobs/job/job_controller_test.go [hunk 1]
```diff
@@ -1209,7 +1166,7 @@ func TestReconciler(t *testing.T) {
 				SetAnnotation(JobMinParallelismAnnotation, "5").
 				Obj(),
 			workloads: []kueue.Workload{
-				*utiltesting.MakeWorkload("a", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
```
[the remark this answers is not in the document]
Comment (yaroslava-serdiuk) [reply pkg/controller/jobs/job/job_controller_test.go:1212]: Why the test didn't fail after this change?

[the remark this answers is not in the document]
Comment (PBundyra) [reply pkg/controller/jobs/job/job_controller_test.go:1212]: > Looks like this change is dropping the finalizer, is it intentional? If not, please put it back.

Thanks for catching that.

> Why the test didn't fail after this change?

This test scenario does not compare workloads so probably that's the reason.

**Pull request merged**